svsim/src/main/scala/verilator/Backend.scala

```diff
@@ -8,9 +8,55 @@ import scala.collection.mutable
 
 object Backend {
   object CompilationSettings {
-    sealed trait TraceStyle
-    object TraceStyle {
-      case class Vcd(traceUnderscore: Boolean = false, filename: String = "") extends TraceStyle
+
+    sealed trait TraceKind {
+      private[Backend] def toCompileFlags: Seq[String]
+    }
+
+    object TraceKind {
+
+      /**
+        * VCD tracing
+        */
+      case object Vcd extends TraceKind {
+        final def toCompileFlags = Seq("--trace")
+      }
+
+      private[Backend] sealed trait FstTraceKind { self: TraceKind =>
+        protected val traceThreads: Int
+        final def toCompileFlags: Seq[String] =
+          Seq("--trace-fst") ++
+            Option.when(traceThreads > 0)(Seq("--trace-threads", traceThreads.toString)).toSeq.flatten
+      }
+
+      /**
+        * FST tracing
+        * 
+        * @param traceThreads If set to a positive value, enables FST waveform creation using `traceThreads` separate threads
+        */
+      case class Fst(traceThreads: Int) extends TraceKind with FstTraceKind
+
+      case object Fst extends TraceKind with FstTraceKind { override val traceThreads = 0 }
```

Comment:
This duplication feels a little baroque. I'd opt for just a single `case class Fst` with a `traceThreads: Option[Int]` parameter.

Reply:
The difference in use with the `Vcd` option didn't feel right, so I resorted to this. But I agree it's a bit weird.
I'll change all `Int` parameters to `Option[Int]` then.